compute_sdk/globus_compute_sdk/sdk/client.py

```diff
@@ -311,11 +313,8 @@ def run(self, *args, endpoint_id=None, function_id=None, **kwargs) -> str:
         task_id : str
         UUID string that identifies the task
         """
-        assert endpoint_id is not None, "endpoint_id key-word argument must be set"
-        assert function_id is not None, "function_id key-word argument must be set"
-
         batch = self.create_batch(endpoint_id)
-        batch.add(function_id, args, kwargs)
+        batch.add(str(function_id), args, kwargs)
```

Comment:
I might prefer changing the `add` signature to accept `UUID_LIKE_T`, since that would also clean up the other call site as well

Reply:
Move the `str()` call one layer deeper.  Works for me.  Done.